Frameworks/QuartzCore/CALayer.mm

```diff
@@ -2477,24 +2484,31 @@ + (CGRect)convertRect:(CGRect)pos fromLayer:(CALayer*)fromLayer toLayer:(CALayer
         return pos;
     }
 
-    if (!fromLayer) {
-        // get the top most layer for the current layer hierarchy
+    UIElement fromLayerElement = nullptr;
+    UIElement toLayerElement = nullptr;
+
+    if (fromLayer) {
+        // Get the backing sublayer xaml UIElement for fromLayer
+        // We use the sublayer element to support proper point conversion within a scrolled view.
+        fromLayerElement = [fromLayer _getSublayerXamlElement];
+    } else if (GetCACompositor()->IsRunningAsFramework()) {
+        // In framework mode we don't own the topmost layer
+        fromLayerElement = Window::Current().Content();
```

Comment:
>fromLayerElement = Window::Current().Content(); [](start = 8, length = 47)

in your description ; "In theory this would also work when not running in framework mode, but I was reluctant to make such a sweeping change without a good reason.". So in theory, this line should work for either cases?